core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -190,13 +198,14 @@ protected void doStop() throws Exception {
 
     List<URI> listeners = parseListeners(config.getList(RestConfig.LISTENERS_CONFIG),
             config.getInt(RestConfig.PORT_CONFIG), Arrays.asList("http", "https"), "http");
+    SslContextFactory sslContextFactory = new SslContextFactory();
+    this.sslContextFactory = sslContextFactory;
     for (URI listener : listeners) {
       log.info("Adding listener: " + listener.toString());
       NetworkTrafficServerConnector connector;
       if (listener.getScheme().equals("http")) {
         connector = new NetworkTrafficServerConnector(server);
       } else {
-        SslContextFactory sslContextFactory = new SslContextFactory();
         if (!config.getString(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG).isEmpty()) {
```

Comment:
is there a reason we are re-configuring the factory for each each ssl listener? I think our settings are not specific to any listener, so it probably makes sense to move this into a separate method and call it once rather than inside a loop